generators/app/templates/service.js

```diff
@@ -23,4 +23,12 @@ module.exports = function() {
   <% if (localAuth || authentication.length) { %>
   app.configure(authentication);<% } %><% for (var i = 0; i < services.length; i++) { %>
   app.configure(<%= services[i] %>);<% } %>
+  <% if (database === 'sqlite' || database === 'mssql' || database === 'postgres' || database === 'mysql' || database === 'mariadb') { %>
+  const {models} = sequelize;
+  app.set('models', models);
+  Object.keys(models).forEach(name => {
+    if ('associate' in models[name]) {
```

Comment:
from `classMethods.associate` in `generators/model/templates/sequelize.js`. This change would include an empty `associate` function on sequelize models which get called after all the models are registered, that way the model load order doesn't matter.

I noticed after making this PR that this same design was already in discussion in #94. They *did* bring up the point though that the model.sync() bits would also need to be moved until *after* these `associate` functions run though.